controllers/provider-alicloud/pkg/controller/worker/machines_test.go

```diff
@@ -283,21 +308,26 @@ var _ = Describe("Machines", func() {
 				}
 
 				scheme = runtime.NewScheme()
-				_ = apisalicloud.AddToScheme(scheme)
-				_ = alicloudv1alpha1.AddToScheme(scheme)
+				_ = api.AddToScheme(scheme)
+				_ = apiv1alpha1.AddToScheme(scheme)
```

Comment:
The more I am reading this PR, the more I do not like how you hide that these packages are containing highly specified code(alicloud in this case).

Reply:
The more I've to change coding using different import labels for the same semantic based on the same coding the more I get crazy just changing packet labels in hundreds of lines of copied coding.

Import labels should express a semantic not a technical import. Therefore the default label is always the last package, not the complete path. The same should be done for explicit labels.
Here, in provider *X* the label `api` always refers to the local api of this provider, regardless of the name of the provider. This significantly simplifies the maintenance of common code in all there providers.

Otherwise you get crazy when copying code among providers. IMHO, it also simplifies the reading of the coding. If I am in provider `X` it is clear that api should refer to my own api. If I want to explicitly use some foreign api, then it makes sense to explicitly label this import according to its main semantic (here provider) to improve the readability. Then it even makes sense to rename the import accordingly even if it would be unique without this renaming.